lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
@dkm2110 Given that batch-processing is the primary use case for IMRU, providing an option for `GC.Collect()` seems to be reasonable here. So long as you're already invoking garbage collection, do you think it's also worth it to investigate how long it takes for a Large-Object-Heap compaction? @markusweimer is right that currently HDInsight container memory limits are not enforced by the CLR, and we've also had to deal with that by periodically calling `GC.Collect()` in ASA after calculating `Process.GetCurrentProcess().PrivateMemorySize64`, unfortunately.